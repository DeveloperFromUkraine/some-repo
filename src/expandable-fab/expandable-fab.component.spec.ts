import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ExpandableFabComponent } from 'index';
import { ComponentTest } from '../../test/test-bed/component';
import { MatIconModule } from '@angular/material';
import { DebugElement } from '@angular/core/src/debug/debug_node';
import { By } from '@angular/platform-browser';

describe('ExpandableFabComponent', () => {
    let fixture: ComponentFixture<ExpandableFabComponent>;
    let component: ExpandableFabComponent;
    let de: DebugElement;
    let ne: HTMLElement;

    beforeEach(async () => {
        ComponentTest.createTestBed([MatIconModule], [ExpandableFabComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(ExpandableFabComponent);
        component = fixture.componentInstance;

        fixture.detectChanges();
    });

    it('should trigger handleClick when click event emitted', () => {
        de = fixture.debugElement.query(By.css('.icon-container'));
        let spy = jest.spyOn(component, 'handleClick');

        de.triggerEventHandler('click', null);

        expect(spy).toHaveBeenCalled;
    });

    it('should set activeClass(null) when handleClick triggered', () => {
        de = fixture.debugElement.query(By.css('.icon-container'));

        de.triggerEventHandler('click', null);

        expect(component.activeClass).toBe('active');
    });

    it('should set activeClass(!null) when handleClick triggered', () => {
        de = fixture.debugElement.query(By.css('.icon-container'));

        component.activeClass = 'not null';
        de.triggerEventHandler('click', null);

        expect(component.activeClass).toBe(null);
    });

    it('should have active classes if isOpen is true', async () => {
        component.isOpen = true;

        await fixture.detectChanges();

        expect(fixture.debugElement.query(By.css('.active'))).toBeTruthy();
        expect(fixture.debugElement.query(By.css('.icon-primary-active'))).toBeTruthy();
        expect(fixture.debugElement.query(By.css('.icon-secondary-active'))).toBeTruthy();
    });

    it('should not have active class if isOpen is false', () => {
        expect(fixture.debugElement.query(By.css('.active'))).not.toBeTruthy();
        expect(fixture.debugElement.query(By.css('.icon-primary-active'))).not.toBeTruthy();
        expect(fixture.debugElement.query(By.css('.icon-secondary-active'))).not.toBeTruthy();
    });
});