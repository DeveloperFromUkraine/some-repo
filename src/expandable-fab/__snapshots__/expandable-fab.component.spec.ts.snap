// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expandable Fab should match snapshot 1`] = `
<ng-component>
  
    <ign-expandable-fab>
    <div
      class="buttons-container active"
      ng-reflect-klass="buttons-container"
      ng-reflect-ng-class="[object Object]"
      role="menu"
    >
      
      
      
      
      
      <ign-expandable-fab-item
        icon="add"
        ng-reflect-icon="add"
      >
        <span
          class="mat-tooltip fab-tooltip"
        >
          
        </span>
        
  
        
        <a
          igna11yclick=""
          mat-fab=""
          role="menuitem"
        >
          
    
          <mat-icon>
            
      add
    
          </mat-icon>
          
  
        </a>
        
  
  
        
        
  
  
      </ign-expandable-fab-item>
      
    
      
      
      
      <div
        class="main-button-container"
        da="mainButton"
      >
        
        
        <a
          aria-expanded="true"
          aria-haspopup="true"
          aria-label=""
          class="icon-container"
          igna11yclick=""
          igna11yviewaccessible=""
          mat-fab=""
          role="button"
        >
          
          
          <mat-icon
            class="icon icon-secondary icon-secondary-active"
            ng-reflect-klass="icon icon-secondary"
            ng-reflect-ng-class="[object Object]"
            style="position:absolute"
          >
            
            clear
          
          </mat-icon>
          
          
          <mat-icon
            class="icon icon-primary icon-primary-active"
            ng-reflect-klass="icon icon-primary"
            ng-reflect-ng-class="[object Object]"
          >
            
            add
          
          </mat-icon>
          
        
        </a>
        
      
      </div>
      
  
    </div>
  </ign-expandable-fab>
</ng-component>
`;
